lib/internal.h

```diff
@@ -349,10 +349,24 @@ static inline int io_getevents(__attribute__((unused)) aio_context_t ctx,
 
 #if __GNUC__ >= 4
 # define DSO_PUBLIC __attribute__ ((visibility ("default")))
+
+# define IMPL_SYMVER(name, version) \
+    __asm__(".global impl_" #name ";"\
+	    ".symver impl_" #name ",kcapi_" #name "@@LIBKCAPI_" version);\
+    __attribute__ ((visibility ("default")))
+
+# define ORIG_SYMVER(name, version) \
+    __asm__(".global orig_" #name ";"\
+	    ".symver orig_" #name ",kcapi_" #name "@LIBKCAPI_" version);\
+    __attribute__ ((visibility ("default")))
+
 #else
 # define DSO_PUBLIC
+# define IMPL_SYMVER(name, version) typedef impl_##name kcapi_##name;
```

Comment:
Doesn't `typedef` work only on types?
https://gcc.godbolt.org/z/fMhc8646W

Reply:
Eh that part I hastly put together, not sure it is correct indeed

Reply:
removed